### PR TITLE
Account management panel

### PR DESCRIPTION
**Pull request by Sam-Jeston**

### CODE REVIEW

File: src/components/AccountDashboard.tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ type Props = ReturnType<typeof mapStateToProps> & typeof mapDispatchToProps
 
 const AccountDashboardPresentation: React.SFC<Props> = props => (
   <div className="container">
+    <AccountPanel />
```
Comment (Carl-Foster) [inline src/components/AccountDashboard.tsx]: Shouldn't this go on the AccountPage?

File: src/containers/AccountPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+              <div className='has-text-centered'>
+                <div className='level'>
+                  <div className='level-item'>
+                    <EditableText
+                      isEditing={this.state.isEditing}
+                      value={this.state.name}
+                      onChangeHandler={this.onChange}
+                      onStartEditing={() => this.setState({isEditing: !this.state.isEditing})}
+                      onStopEditing={this.onStopEditing}
+                      opts={{isLarge: true}}
```
Comment (Carl-Foster) [inline src/containers/AccountPanel.tsx]: Not a big fan of the shorthand

File: src/containers/Notifier.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { connect } from 'react-redux'
+
+const mapStateToProps = ({ notifier }: RootState) => ({ ...notifier })
+
+const mapDispatchToProps = { clearNotification }
+
+type Props = typeof mapDispatchToProps & ReturnType<typeof mapStateToProps>
+
+const NotifierComponent: React.SFC<Props> = (props) => {
+  return (
```
Comment (Carl-Foster) [inline src/containers/Notifier.tsx]: Can get rid of the return by just having the parentheses immediately after the arrow

File: src/containers/__tests__/AccountPanel.test.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import { shallow } from 'enzyme'
+import { Keypair } from 'js-kinesis-sdk'
+import * as React from 'react'
+import '../../setupTests'
+
+describe.only('AccountPanel', () => {
+  let props
+
+  beforeEach(() => {
+    props = {
```
Comment (Carl-Foster) [inline src/containers/__tests__/AccountPanel.test.tsx]: This means you don't get any type errors for props being passed in this file. Unsure if it's intended. Don't think it's a great idea and can save time if just defined as a const above (as you ran into issues with it).

File: src/containers/__tests__/AccountPanel.test.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    const localProps = { ...props, showNotification, updateAccountName }
+
+    const wrapper = shallow(<AccountPanelComponent {...localProps} />)
+    const instance = wrapper.instance() as AccountPanelComponent
+
+    instance.onChange({target: {value: 'c'}})
+    instance.onStopEditing()
+
+    expect(showNotification.mock.calls.length).toBe(1)
+    expect(updateAccountName.mock.calls.length).toBe(0)
```
Comment (Carl-Foster) [inline src/containers/__tests__/AccountPanel.test.tsx]: I think there's the helper `toBeCalledOnce()` and `toNotBeCalled()` (or something like that)

File: src/store/reducers/wallet.ts [hunk 6]
```diff
@@ -17,6 +17,16 @@ interface WalletState extends WalletLoggedInState {
...
 }
 
+function accountNameStateChange<T extends BaseAccount>(state: any[], {existingName, newName}: {existingName: string, newName: string}): T[] {
+  const existingAccount = state.find(a => a.name === existingName)
+  const remainingAccounts = state.filter(a => a.name !== existingName)
+  const newAccount = {
+    ...existingAccount,
+    name: newName
+  }
+  return [...remainingAccounts, newAccount]
```
Comment (Carl-Foster) [inline src/store/reducers/wallet.ts]: This could be a `state.map()` call

Comment (Carl-Foster) [reply src/store/reducers/wallet.ts]: `state.map(a => a.name === existingName ? {...a, name: newName} : a)`

File: src/components/EditableText.tsx [hunk 7]
```diff
@@ -4,22 +4,26 @@ interface EditableProps {
...
+          style={{textTransform: 'initial'}}
+          className={this.inputClasses() + (isLarge ? ' is-large' : '')}
           value={this.props.value}
           onChange={this.props.onChangeHandler}
           onBlur={this.props.onStopEditing}
           onClick={this.onClick}
         />
         {!this.props.isEditing && (
-          <span className="icon is-right">
+          <span className="icon is-right" style={isLarge ? {fontSize: '24px'} : {}}>
```
Comment (desnor) [inline src/components/EditableText.tsx]: I would try adding the `is-medium` or `is-large` class onto the span and/or `fa-lg` class onto the icon itself, in place of changing the font size directly, eg
```typescript
  <span className="icon is-right is-large" >
    <i className="fa fa-lg fa-edit" />
  </span>
```

File: src/components/Sign.tsx [hunk 8]
```diff
@@ -97,7 +97,7 @@ export class SignForm extends React.Component<SignProps, State> {
     const { signData, handleSignFormChange, signature } = this.props
     const { copied } = this.state
     return (
-      <div className="is-centered">
+      <div>
```
Comment (desnor) [inline src/components/Sign.tsx]: can we take out the wrapping div?

**Pull request merged**